### PR TITLE
Add json.net dependency, fix incremental build

### PR DESCRIPTION
**Pull request by BillHiebert**
Small check in which adds a dependency on Json.Net  8.0.3. Also fixes the incremental compilation problem.

@abpiskunov @davkean

### CODE REVIEW

File: build/Targets/ProducesNoOutput.Imports.targets [hunk 1]
```diff
@@ -7,8 +7,9 @@
     <TargetPath></TargetPath>    <!-- Prevent projects referencing this from trying to pass us to the compiler -->
   </PropertyGroup>
   
-  <Target Name="CoreCompile" />                               <!-- Prevent Csc from being called -->  
-  <Target Name="GenerateTargetFrameworkMonikerAttribute" />   <!-- Don't generate TFM attribute -->  
+  <!-- Return this back when up-to-date bug is fixed (projects are being build everytime) -->
```
Comment (srivatsn) [inline build/Targets/ProducesNoOutput.Imports.targets]: Is there a bug number tracking this?

File: build/Targets/References.targets [hunk 2]
```diff
@@ -78,7 +78,10 @@
           <Project>{e6796b97-d5c6-45b2-ae46-351d15dcfc71}</Project>
           <Name>Metadata</Name>
         </ProjectReference>
-        
+        <ProjectReference Include="$(MSBuildThisFileDirectory)..\..\src\Dependencies\Json.net\Json.net.csproj">
```
Comment (srivatsn) [inline build/Targets/References.targets]: I assume this is needed for reading launch profiles?

**Comment by srivatsn** [comment]
👍

**Comment by davkean** [comment]
:+1:

File: build/Targets/ProducesNoOutput.Imports.targets (hunk 1 above)
Comment (BillHiebert) [inline build/Targets/ProducesNoOutput.Imports.targets]: Incremental build is covered by: https://github.com/dotnet/roslyn-project-system/issues/62

**Pull request merged**